sdk/spring/azure-spring-boot/src/main/java/com/azure/spring/keyvault/KeyVaultEnvironmentPostProcessorHelper.java

```diff
@@ -183,4 +184,23 @@ private String getPropertyValue(final String normalizedName, final Property prop
             .map(environment::getProperty)
             .orElse(defaultValue);
     }
+
+    private String getPropertyValueFromAzureProperties(final Property property, String defaultValue) {
+        return Optional.of(KeyVaultProperties.getPropertyNameFromAzureProperties(property))
+            .map(environment::getProperty)
+            .orElse(defaultValue);
+    }
+
+    private String getProperty(final String normalizedName, final Property property) {
+        return getProperty(normalizedName, property, null);
+    }
+
+    private String getProperty(final String normalizedName, final Property property, String defaultValue) {
+        if (normalizedName != "") {
+            return getPropertyValue(normalizedName, property, defaultValue);
+        }
+
+        return Optional.ofNullable(getPropertyValue(normalizedName, property, defaultValue))
+            .orElse(getPropertyValueFromAzureProperties(property, defaultValue));
```

Comment:
Think about this case:
1. defaultValue = `defaultValue1`.
2. getPropertyValue(normalizedName, property, null) return `null`.
3. getPropertyValueFromAzureProperties(property, null) return `propertyValueFromAzureProperties1`.

In this case, I think we should return `propertyValueFromAzureProperties1`, but current code will return `defaultValue1`.

And current code is a bit complex, could you please:
1. Simplify current code.
2. Add unit test for it.